### PR TITLE
Update model card for Depth Anything

### PR DESCRIPTION
**Pull request by shubham0204**
# What does this PR do?

This PR updates the model-card for the `depth_anything` model, as described in #36979, in an attempt to standardize all model-cards.

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@stevhliu

### CODE REVIEW

File: docs/source/en/model_doc/depth_anything.md [hunk 1]
```diff
@@ -14,101 +14,82 @@ rendered properly in your Markdown viewer.
...
+[Depth Anything](https://huggingface.co/papers/2401.10891) is a monocular depth estimation (MDE) model that is trained on a large number of unlabelled images with a self-supervised learning approach. It uses a pre-trained [DINOv2](./dinov2) model as an image encoder and the [DPT](./dpt) architecture for the decoder. A 'teacher' model is trained on manual labels from from standard datasets whereas the 'student' model is trained on a combination of, pseudo-labels predicted by the 'teacher' model and the manual labels. To enhance the generalization capability of the 'student' model, its training inputs are strongly perturbed.
```
Comment (stevhliu) [inline docs/source/en/model_doc/depth_anything.md:25]:
```suggestion
[Depth Anything](https://huggingface.co/papers/2401.10891) is designed to be a foundation model for monocular depth estimation (MDE). It is jointly trained on labeled and ~62M unlabeled images to enhance the dataset. It uses a pretrained [`DINOv2`] model as an image encoder to inherit its existing rich semantic priors, and [`DPT`] as the decoder. A teacher model is trained on unlabeled images to create pseudo-labels. The student model is trained on a combination of the pseudo-labels and labeled images. To improve the student model's performance, strong perturbations are added to the unlabeled images to challenge the student model to learn more visual knowledge from the image.
```

File: docs/source/en/model_doc/depth_anything.md [hunk 2]
```diff
@@ -14,101 +14,82 @@ rendered properly in your Markdown viewer.
...
 
 <small> Depth Anything overview. Taken from the <a href="https://arxiv.org/abs/2401.10891">original paper</a>.</small>
 
-This model was contributed by [nielsr](https://huggingface.co/nielsr).
-The original code can be found [here](https://github.com/LiheYoung/Depth-Anything).
-
-## Usage example
-
-There are 2 main ways to use Depth Anything: either using the pipeline API, which abstracts away all the complexity for you, or by using the `DepthAnythingForDepthEstimation` class yourself.
+The checkpoints for Depth Anything can be found in [Depth Anything Release](https://huggingface.co/collections/LiheYoung/depth-anything-release-65b317de04eec72abf6b55aa) collection by one of its authors [Lihe Yang](https://huggingface.co/LiheYoung).
```
Comment (stevhliu) [inline docs/source/en/model_doc/depth_anything.md:32]:
```suggestion
You can find all the original Depth Anything checkpoints under the [Depth Anything](https://huggingface.co/collections/LiheYoung/depth-anything-release-65b317de04eec72abf6b55aa) release.
```

File: docs/source/en/model_doc/depth_anything.md [hunk 3]
```diff
@@ -14,101 +14,82 @@ rendered properly in your Markdown viewer.
...
 
-### Pipeline API
+> [!TIP]
+> Click on the Depth Anything models in the right sidebar for more examples of how to apply Gemma to different vision and language tasks.
```
Comment (stevhliu) [inline docs/source/en/model_doc/depth_anything.md:35]:
```suggestion
> Click on the Depth Anything models in the right sidebar for more examples of how to apply Depth Anything to different vision tasks.

The example below demonstrates how to obtain a depth map with [`Pipeline`] or the [`AutoModel`] class.
```

File: docs/source/en/model_doc/depth_anything.md [hunk 4]
```diff
@@ -14,101 +14,82 @@ rendered properly in your Markdown viewer.
...
-The pipeline allows to use the model in a few lines of code:
+<hfoptions id="usage">
+<hfoption id="Pipeline">
 
-```python
->>> from transformers import pipeline
->>> from PIL import Image
->>> import requests
+```py
+from transformers import pipeline
```
Comment (stevhliu) [inline docs/source/en/model_doc/depth_anything.md:41]: Even shorter :)

```py
import torch
from transformers import pipeline

pipe = pipeline(task="depth-estimation", model="LiheYoung/depth-anything-base-hf", torch_dtype=torch.bfloat16, device=0)
pipe("http://images.cocodataset.org/val2017/000000039769.jpg")["depth"]
```

File: docs/source/en/model_doc/depth_anything.md [hunk 5]
```diff
@@ -14,101 +14,82 @@ rendered properly in your Markdown viewer.
...
-... )
-
->>> predicted_depth = post_processed_output[0]["predicted_depth"]
->>> depth = (predicted_depth - predicted_depth.min()) / (predicted_depth.max() - predicted_depth.min())
->>> depth = depth.detach().cpu().numpy() * 255
->>> depth = Image.fromarray(depth.astype("uint8"))
+</hfoption>
+<hfoption id="AutoModel">
+
+```py
```
Comment (stevhliu) [inline docs/source/en/model_doc/depth_anything.md:60]: Cleaned it up a bit to be consistent

```py
import torch
import requests
import numpy as np
from PIL import Image
from transformers import AutoImageProcessor, AutoModelForDepthEstimation

image_processor = AutoImageProcessor.from_pretrained("LiheYoung/depth-anything-base-hf")
model = AutoModelForDepthEstimation.from_pretrained("LiheYoung/depth-anything-base-hf", torch_dtype=torch.bfloat16)
url = "http://images.cocodataset.org/val2017/000000039769.jpg"
image = Image.open(requests.get(url, stream=True).raw)
inputs = image_processor(images=image, return_tensors="pt")

with torch.no_grad():
    outputs = model(**inputs)

post_processed_output = image_processor.post_process_depth_estimation(
    outputs,
    target_sizes=[(image.height, image.width)],
)
predicted_depth = post_processed_output[0]["predicted_depth"]
depth = (predicted_depth - predicted_depth.min()) / (predicted_depth.max() - predicted_depth.min())
depth = depth.detach().cpu().numpy() * 255
Image.fromarray(depth.astype("uint8"))
```

File: docs/source/en/model_doc/depth_anything.md [hunk 6]
```diff
@@ -14,101 +14,82 @@ rendered properly in your Markdown viewer.
...
-A list of official Hugging Face and community (indicated by 🌎) resources to help you get started with Depth Anything.
+</hfoption>
+</hfoptions>
 
-- [Monocular depth estimation task guide](../tasks/monocular_depth_estimation)
-- A notebook showcasing inference with [`DepthAnythingForDepthEstimation`] can be found [here](https://github.com/NielsRogge/Transformers-Tutorials/blob/master/Depth%20Anything/Predicting_depth_in_an_image_with_Depth_Anything.ipynb). 🌎
+## Notes
 
-If you're interested in submitting a resource to be included here, please feel free to open a Pull Request and we'll review it! The resource should ideally demonstrate something new instead of duplicating an existing resource.
+- [Depth Anything V2](depth_anything_v2), released in June 2024, uses the same architecture as Depth Anything and therefore is compatible with all code examples and existing workflows. However, it leverages synthetic data and a larger capacity teacher model to achieve much finer and robust depth predictions.
```
Comment (stevhliu) [inline docs/source/en/model_doc/depth_anything.md:92]:
```suggestion
- [`DepthAnythingV2`], released in June 2024, uses the same architecture as Depth Anything and is compatible with all code examples and existing workflows. It uses synthetic data and a larger capacity teacher model to achieve much finer and robust depth predictions.
```

**Review by stevhliu — COMMENTED** [review]
Super nice, just some nits to keep it consistent with the existing model cards. Thanks again! 🤗

**Comment by shubham0204** [comment]
@stevhliu I have made all suggested changes and amended the last commit.

File: docs/source/en/model_doc/depth_anything.md [hunk 7]
```diff
@@ -14,101 +14,74 @@ rendered properly in your Markdown viewer.
...
+[Depth Anything](https://huggingface.co/papers/2401.10891) is designed to be a foundation model for monocular depth estimation (MDE). It is jointly trained on labeled and ~62M unlabeled images to enhance the dataset. It uses a pretrained [`DINOv2`] model as an image encoder to inherit its existing rich semantic priors, and [`DPT`] as the decoder. A teacher model is trained on unlabeled images to create pseudo-labels. The student model is trained on a combination of the pseudo-labels and labeled images. To improve the student model's performance, strong perturbations are added to the unlabeled images to challenge the student model to learn more visual knowledge from the image.
```
Comment (stevhliu) [inline docs/source/en/model_doc/depth_anything.md:25]:
```suggestion
[Depth Anything](https://huggingface.co/papers/2401.10891) is designed to be a foundation model for monocular depth estimation (MDE). It is jointly trained on labeled and ~62M unlabeled images to enhance the dataset. It uses a pretrained [DINOv2](./dinov2) model as an image encoder to inherit its existing rich semantic priors, and [DPT](./dpt) as the decoder. A teacher model is trained on unlabeled images to create pseudo-labels. The student model is trained on a combination of the pseudo-labels and labeled images. To improve the student model's performance, strong perturbations are added to the unlabeled images to challenge the student model to learn more visual knowledge from the image.
```

File: docs/source/en/model_doc/depth_anything.md [hunk 8]
```diff
@@ -14,101 +14,74 @@ rendered properly in your Markdown viewer.
...
 
 <img src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/transformers/model_doc/depth_anything_overview.jpg"
```
Comment (stevhliu) [inline docs/source/en/model_doc/depth_anything.md:27]: We can remove image and the `<small>` caption under it

File: docs/source/en/model_doc/depth_anything.md [hunk 9]
```diff
@@ -14,101 +14,74 @@ rendered properly in your Markdown viewer.
...
 
 <small> Depth Anything overview. Taken from the <a href="https://arxiv.org/abs/2401.10891">original paper</a>.</small>
 
-This model was contributed by [nielsr](https://huggingface.co/nielsr).
-The original code can be found [here](https://github.com/LiheYoung/Depth-Anything).
-
-## Usage example
-
-There are 2 main ways to use Depth Anything: either using the pipeline API, which abstracts away all the complexity for you, or by using the `DepthAnythingForDepthEstimation` class yourself.
+You can find all the original Depth Anything checkpoints under the [Depth Anything](https://huggingface.co/collections/LiheYoung/depth-anything-release-65b317de04eec72abf6b55aa) release.
```
Comment (stevhliu) [inline docs/source/en/model_doc/depth_anything.md:32]:
```suggestion
You can find all the original Depth Anything checkpoints under the [Depth Anything](https://huggingface.co/collections/LiheYoung/depth-anything-release-65b317de04eec72abf6b55aa) collection.
```

File: docs/source/en/model_doc/depth_anything.md [hunk 10]
```diff
@@ -14,101 +14,74 @@ rendered properly in your Markdown viewer.
...
-A list of official Hugging Face and community (indicated by 🌎) resources to help you get started with Depth Anything.
+</hfoption>
+</hfoptions>
 
-- [Monocular depth estimation task guide](../tasks/monocular_depth_estimation)
-- A notebook showcasing inference with [`DepthAnythingForDepthEstimation`] can be found [here](https://github.com/NielsRogge/Transformers-Tutorials/blob/master/Depth%20Anything/Predicting_depth_in_an_image_with_Depth_Anything.ipynb). 🌎
+## Notes
 
-If you're interested in submitting a resource to be included here, please feel free to open a Pull Request and we'll review it! The resource should ideally demonstrate something new instead of duplicating an existing resource.
+- [`DepthAnythingV2`], released in June 2024, uses the same architecture as Depth Anything and is compatible with all code examples and existing workflows. It uses synthetic data and a larger capacity teacher model to achieve much finer and robust depth predictions.
```
Comment (stevhliu) [inline docs/source/en/model_doc/depth_anything.md:84]:
```suggestion
- [DepthAnythingV2](./depth_anything_v2), released in June 2024, uses the same architecture as Depth Anything and is compatible with all code examples and existing workflows. It uses synthetic data and a larger capacity teacher model to achieve much finer and robust depth predictions.
```

**Review by stevhliu — APPROVED** [review]
Great! Just a few more changes and we can merge :)

**Comment by shubham0204** [comment]
@stevhliu all suggested changes have been made in the latest commit 👍🏻
